AK Apolo/app/src/main/java/ayds/apolo/songinfo/moredetails/fulllogic/ArtistDescriptionActivity.kt

```diff
@@ -0,0 +1,163 @@
+import android.net.Uri
+import android.os.Build
+import com.squareup.picasso.Picasso
+import android.text.Html
+import android.text.Spanned
+import android.view.View
+import android.widget.ImageView
+import com.google.gson.JsonElement
+import retrofit2.Response
+import java.io.IOException
+import java.lang.StringBuilder
+import java.util.*
+
+internal const val CONTENT = "content"
+internal const val URL = "url"
+internal const val BIO = "bio"
+internal const val URL_AUDIO_SCROBBLER = "https://ws.audioscrobbler.com/2.0/"
+internal const val LASTFM_IMAGE_WIKIPEDIA = "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d4/Lastfm_logo.svg/320px-Lastfm_logo.svg.png"
+internal const val WIDTH_DESCRIPTION = 400
+
+class ArtistDescriptionActivity: AppCompatActivity() {
+
+    private var infoArtistTextView: TextView? = null
+    private val artistDataBase: ArtistDataBase = ArtistDataBase(this)
+
+    private var urlLastFM: String? = ""
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_other_info)
+        infoArtistTextView = findViewById(R.id.textPane2)
+
+        showArtistInfo(intent.getStringExtra(ARTIST_NAME_EXTRA) ?: "")
+
+        setUrlButtonListener()
+    }
+
+    private fun setUrlButtonListener() {
+        findViewById<View>(R.id.openUrlButton).setOnClickListener {
+            val intent = Intent(Intent.ACTION_VIEW)
+            intent.data = Uri.parse(urlLastFM)
+            startActivity(intent)
+        }
+    }
+
+    private fun showArtistInfo(artistName: String) {
+        Thread {
+            val infoArtist = getInfoArtist(artistName)
+
+            setArtistInfoOnUI(infoArtist)
+        }.start()
+    }
+
+    private fun getInfoArtist(artistName: String): String {
+        var infoArtist: String? = getInfoArtistFromDB(artistName)
+
+        infoArtist = if (infoArtist == null) {
+            val infoArtistLastFM = getInfoArtistFromLastFM(artistName)
+            artistDataBase.saveArtist(artistName, infoArtistLastFM)
+
+            infoArtistLastFM
+        } else
+            "[*]$infoArtist"
+
+        return infoArtist
+    }
+
+    private fun getInfoArtistFromLastFM(artistName: String): String {
+        return try {
+            val lastFMResponse = getInfoArtistFromLastFMAPI(createLastFMAPI(), artistName)
+            val extractInfoArtist = extractInfoFromResponse(lastFMResponse) ?: return ""
+
+            getArtistNameDescription(artistName, extractInfoArtist)
+        } catch (e1: IOException) {
+            e1.printStackTrace()
+
+            ""
+        }
+    }
+
+    private fun getInfoArtistFromLastFMAPI(lastFMAPI: LastFMAPI, artistName: String) =
+        lastFMAPI.getArtistInfo(artistName).execute()
+
+    private fun extractInfoFromResponse(callResponse: Response<String>): JsonElement? {
+        val artist = getArtistJSON(callResponse)
+        val bio = artist[BIO]?.asJsonObject
+        val url = artist[URL]
```

Comment:
No es necesaria la variable url
```suggestion
        urlLastFm = artist[URL]?.asString
```